dhis-2/dhis-test-web-api/src/test/java/org/hisp/dhis/test/webapi/AssertionsTest.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2004-2025, University of Oslo
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.test.webapi;
+
+import static org.hisp.dhis.test.webapi.Assertions.assertNoDiff;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrowsExactly;
+
+import org.junit.jupiter.api.Test;
+import org.opentest4j.AssertionFailedError;
+
+class AssertionsTest {
+
+  @Test
+  void testAssertNoDiff() {
+
+    AssertionFailedError ex =
+        assertThrowsExactly(
+            AssertionFailedError.class,
+            () ->
+                assertNoDiff(
+                    """
+                { "page": 1, "pageSize": 50 }""",
+                    """
+                { "paga": 1, "pageSize": 50 }"""));
+    assertEquals(
+        """
+        JSON has 2 structural differences:
+          ++ $.paga: ? <> 1
```

Comment:
What does `? <> 1` and its reverse below mean?

Reply:
`?` is "not existing"/"nothing", `<>` is "differs from"/"not equal to" (`!=` was already taken in the context)

Reply:
and why is `nothing is different from 1` showing up on an unexpected key?

and the reverse is then `1 is different from nothing` 😅